### PR TITLE
MLT-0032 Tests for OrderService

### PR DESCRIPTION
**Pull request by anotheroneofthese**
Adds a some tests for OrderService

Has one breaking change with renaming `getByHostNameAndHostOrderId` to `findByHostNameAndHostOrderId` in OrderRepository, to be more consistent with ProductRepository

### CODE REVIEW

File: src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt [hunk 1]
```diff
@@ -35,7 +38,12 @@ class OrderService(val db: OrderRepository, val synqService: SynqOrderService) {
             return ResponseEntity.badRequest().build()
         }
 
-        synqService.createOrder(payload.toOrder().toSynqPayload())
+        // Handle any edge-cases
+        val synqResponse = synqService.createOrder(payload.toOrder().toSynqPayload())
+        if (!synqResponse.statusCode.isSameCodeAs(HttpStatus.CREATED)) {
+            throw ServerErrorException("Unexpected error with SynQ", null)
```
Comment (MormonJesus69420) [inline src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt:44]: Need to talk about it with Tom, how should we handle it

File: src/test/kotlin/no/nb/mlt/wls/order/service/OrderServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        assertExceptionThrownWithMessage(top.copy(orderId = "\t\n"), "orderId is required", ServerWebInputException::class.java)
+        assertExceptionThrownWithMessage(top.copy(orderId = "      "), "orderId is required", ServerWebInputException::class.java)
+    }
+
+    @Test
+    fun `save called with payload missing hostOrderId throws`() {
+        assertExceptionThrownWithMessage(top.copy(hostOrderId = ""), "The order's hostOrderId is required", ServerWebInputException::class.java)
+        assertExceptionThrownWithMessage(top.copy(hostOrderId = "\t\n"), "The order's hostOrderId is required", ServerWebInputException::class.java)
+        assertExceptionThrownWithMessage(top.copy(hostOrderId = "      "), "The order's hostOrderId is required", ServerWebInputException::class.java)
+    }
```
Comment (MormonJesus69420) [inline src/test/kotlin/no/nb/mlt/wls/order/service/OrderServiceTest.kt:56]: What about orders with no productLines?

File: src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt (hunk 1 above)
Comment (anotheroneofthese) [reply src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt:44]: To clarify this edge-case only happens if the `createOrder` call doesn't throw, but still returns something that isn't a new order. Will update the comment accordingly

File: src/test/kotlin/no/nb/mlt/wls/order/service/OrderServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        runTest {
+            every { db.findByHostNameAndHostOrderId(top.hostName, top.hostOrderId) } returns Mono.empty()
+            coEvery { synq.createOrder(any()) } throws ServerErrorException("Duplicate order found in in SynQ", null)
+
+            assertExceptionThrownWithMessage(top, "Duplicate order", ServerErrorException::class.java)
+        }
+    }
+
+    @Test
+    fun `save called when synq fails is handled gracefully`() {
```
Comment (MormonJesus69420) [inline src/test/kotlin/no/nb/mlt/wls/order/service/OrderServiceTest.kt:77]: It's mister SynQ for you, son

**Review by MormonJesus69420 — CHANGES_REQUESTED** [review]

File: src/test/kotlin/no/nb/mlt/wls/order/service/OrderServiceTest.kt (hunk 2 above)
Comment (anotheroneofthese) [reply src/test/kotlin/no/nb/mlt/wls/order/service/OrderServiceTest.kt]: This now has a test

File: src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt (hunk 1 above)
Comment (MormonJesus69420) [reply src/main/kotlin/no/nb/mlt/wls/order/service/OrderService.kt:45]: Yeah, just thinking about what to give back to caller and how to handle it in DB and storage systems, eg cleanup
